File: tests/src/com.fasterxml.jackson.module/jackson-module-scala_3/2.15.0/src/test/scala/com_fasterxml_jackson_module/jackson_module_scala_3/EnumResolverTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_fasterxml_jackson_module.jackson_module_scala_3

import com.fasterxml.jackson.core.`type`.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.scala.DefaultScalaModule
import com.fasterxml.jackson.module.scala.JsonScalaEnumeration
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

import scala.annotation.meta.field

class EnumResolverTest {
  private val mapper: ObjectMapper = new ObjectMapper().registerModule(DefaultScalaModule)

  @Test
  def deserializesAnnotatedEnumerationValueFromString(): Unit = {
    val value: EnumResolverEnvelope =
      mapper.readValue("""{"state":"Published"}""", classOf[EnumResolverEnvelope])

    assertThat(value.state).isEqualTo(EnumResolverFixture.Published)
  }
}

object EnumResolverFixture extends Enumeration {
  val Draft: Value = Value("Draft")
  val Published: Value = Value("Published")
}

class EnumResolverFixtureType extends TypeReference[EnumResolverFixture.type]

class EnumResolverEnvelope {
  @(JsonScalaEnumeration @field)(classOf[EnumResolverFixtureType])
  var state: EnumResolverFixture.Value = EnumResolverFixture.Draft
}

File: tests/src/org.testcontainers/testcontainers/1.19.8/src/test/java/org_testcontainers/testcontainers/EnumResolverTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_testcontainers.testcontainers;

import com.fasterxml.jackson.annotation.JsonValue;
import org.junit.jupiter.api.Test;
import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;

import static org.assertj.core.api.Assertions.assertThat;

public class EnumResolverTest {
    @Test
    void resolvesEnumConstantsUsingAJsonValueMethod() throws Exception {
        ObjectMapper mapper = new ObjectMapper();

        assertThat(mapper.readValue("\"active-code\"", State.class)).isEqualTo(State.ACTIVE);
    }

    public enum State {
        ACTIVE("active-code"),
        IDLE("idle-code");

        private final String code;

        State(String code) {
            this.code = code;
        }

        @JsonValue
        public String code() {
            return code;
        }
    }
}
